src/components/Button/Button.jsx

```diff
@@ -39,96 +40,98 @@ const Button = forwardRef(({ children, icon, direction, caption, captionColor, d
   </Base>
 ))
 
-const colorVariants = variant({
-  default: 'filled',
-  key: 'button',
-  variants: {
-    filled: css`
-      background-color: ${({ color }) => th.color(color)};
-      border-color: ${({ color }) => th.color(color)};
-      color: white;
-      &:hover {
-        background-color: ${({ color }) => th.color(`${color}_hover`)};
-        border-color: ${({ color }) => th.color(`${color}_hover`)};
-      }
-      &:active {
-        background-color: ${({ color }) => th.color(`${color}_active`)};
-        border-color: ${({ color }) => th.color(`${color}_active`)};
-      }
-      &:disabled {
-        background-color: disabled;
-        border-color: disabled;
-        color: white;
-      }
-    `,
-    outlined: css`
-      ${StyledIcon} {
-        fill: ${({ color }) => th.color(color)};
-        &:hover {
-          fill: ${({ color }) => th.color(`${color}_hover`)};
+const colorVariants = ({ theme: { colors }, color }) =>
+  variant({
+    prop: 'variant',
+    variants: {
+      filled: {
+        backgroundColor: colors[color],
+        borderColor: colors[color],
+        color: 'white',
+        '&:hover': {
+          backgroundColor: `${colors[`${color}_hover`]}`,
```

Comment:
Acho que da pra tirar mais esse template string

```suggestion
          backgroundColor: colors[`${color}_hover`],
```

_**obs: se bombar, replica nos demais!**_